File: Java/Stacks/src/FourthExerciseProgram.java
import java.lang.reflect.Array;

public class FourthExerciseProgram {
    private static final Stack<Integer> binary = new Stack<Integer>(10);
    public FourthExerciseProgram(){
    }

    private static Object[] exec(int num){
        int castedNum = num;

        binary.push(castedNum % 2);

        if(castedNum == 1 || castedNum == 0){
            return binary.invert();
        }

        return exec( num / 2);
    }

    public String getBinaryOf(int num) {
        String bin = "";
        Object[] ourNum = exec(num);
        for(int i = 0; i < ourNum.length - 1; i++){
            if(ourNum[i] != null){
                bin += ourNum[i];
            }
        }
        return bin;
    }


}

File: Kotlin/Stacks/src/FourthExerciseProgram.kt
import java.math.BigInteger
import kotlin.math.roundToInt
import java.util.Arrays

class FourthExerciseProgram {
    private val binary = Stack<Int>(10)

    private fun exec(num: Number) : Array<Any> {
        val castedNum = num.toInt()

        binary.push(castedNum % 2)
        //breakpoint
        if(castedNum == 1 || castedNum == 0){
            return binary.invert()
        }

        return exec(num.toInt() / 2)
    }

    fun getBinaryOf(num: Double) : String {
        var bin: String = ""
        exec(num).forEach {
            if(it !is Unit){
                bin += it.toString()
            }
        }

        return bin
    }

}